update mint LSP7 token (#738)
lukso-network/docs: docs/learn/smart-contract-developers/mint-lsp7-token.md

# Mint LSP7 Token

> In this guide you will mint some [LSP7 Digital Asset](../../standards/tokens/LSP7-Digital-Asset.md) tokens to the EOA that controls the contract.

Reviewer: ```suggestion
This guide will teach you how to mint [LSP7 Digital Asset](../../standards/tokens/LSP7-Digital-Asset.md) tokens as an EOA contract owner.
```
"Controls" might be misleading in this context. I would suggest updating the sentence for clarity.

Author: Applied the second part, left the first part as it is. I remember discussing with Fabian about not using "teaching"

## Setup

> The following code snippets require to install a few libraries.

Reviewer: ```suggestion
The following code snippets require the installation a the following libraries:
```

- [`ethers.js`](https://github.com/ethers-io/ethers.js/)
- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)

```shell
npm install ethers @lukso/lsp-smart-contracts
```

### Imports and constants

> At this point, the `LPS7Mintable` contract is being prepared for the following interaction. You construct an instance of a contract, using its _ABI_ and the _contract address_.

Reviewer: ```suggestion
After defining the core parameters of the [`LPS7 Mintable`](../../contracts/contracts/LSP7DigitalAsset/presets/LSP7Mintable.md) contract, you are able to create an instance using its _ABI_ and the _contract address_.
```
What does "the following interaction" mean? Is it necessary. The above suggestion would be my approach.

```javascript
import LSP7Mintable from '@lukso/lsp-smart-contracts/artifacts/LSP7Mintable.json';
import { ethers } from 'ethers';

const privateKey = '0x...';
const myTokenAddress = '0x...';
const universalProfileAddress = '0x...';

const provider = new ethers.JsonRpcProvider('https://rpc.testnet.lukso.network');

// setup signer
const signer = new ethers.Wallet(privateKey as string, provider);
console.log('🔑 EOA: ', signer.address);
```

### Instantiate the contracts

At this point, the `LPS7Mintable` contract is being prepared for the following intercation. you construct an instance of a contract, using _contract ABI_ and _contract address_.

```javascript
let myToken = new ethers.Contract(myTokenAddress, LSP7Mintable.abi, signer);
```

> ### Send transaction

Reviewer: ```suggestion
### Send the transaction
```

Not in polish language, let's put an article 😊

> Finally, you send the transaction and mint some tokens.

Reviewer: ```suggestion
Finally, you can send the transaction to mint tokens.
```

:::warning

> The example contract here allows only the owner of the contract to mint Digital Assets. There might be contracts that don't have this _requirement_.

Reviewer: ```suggestion
The sample contract of this guide only allows the smart contract owner to mint assets. Custom [LSP7](../../standards/tokens/LSP7-Digital-Asset.md) implementations might implement different permission sets.  
```

# Mint LSP7 Token

In this guide you will mint some [LSP7 Digital Asset](../../standards/tokens/LSP7-Digital-Asset.md) tokens to the EOA owner of your token.

## Setup

The following code snippets require to install a few libraries.

- [`ethers.js`](https://github.com/ethers-io/ethers.js/)
- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)

```shell
npm install ethers @lukso/lsp-smart-contracts
```

### Imports and constants

At this point you will need a private key in order to mint some tokens as well as the `LSP7Mintable` _token contract address_.
You will import `LSP7Mintable` in order to get the _ABI_ of the contract that you will interact with.

```javascript
import LSP7Mintable from '@lukso/lsp-smart-contracts/artifacts/LSP7Mintable.json';
import { ethers } from 'ethers';

const privateKey = '0x...';
const myTokenAddress = '0x...';
const universalProfileAddress = '0x...';

const provider = new ethers.JsonRpcProvider('https://rpc.testnet.lukso.network');

// setup signer

> const signer = new ethers.Wallet(privateKey as string, provider);

Author: I believe this is standard with ethers v5 (which is what comes with hardhat) 
https://docs.ethers.org/v5/api/signer/#Wallet

console.log('🔑 EOA: ', signer.address);
```

### Instantiate the contracts

At this point, the `LPS7Mintable` contract is being prepared for the following intercation. you construct an instance of a contract, using _contract ABI_ and _contract address_.

```javascript
let myToken = new ethers.Contract(myTokenAddress, LSP7Mintable.abi, signer);
```

### Send transaction

Finally, you send the transaction and mint some tokens.

:::warning

The example contract here allows minting Digital Assets **only to the owner** of the contract. There might be contracts that don't have this _requirement_.

:::

```javascript
let mintTxn = await myToken.mint(signer.address, 1, true, '0x', {
  gasLimit: 400_000,
});
```

### Final code

```javascript
import { ethers } from 'hardhat';
import LSP7Mintable from '@lukso/lsp-smart-contracts/artifacts/LSP7Mintable.json';

const privateKey = '0x..';
const myTokenAddress = '0x..';

async function main() {

  const provider = new ethers.JsonRpcProvider(
    'https://rpc.testnet.lukso.network',
  );

  const signer = new ethers.Wallet(privateKey as string, provider);
  console.log('🔑 EOA: ', signer.address);

>   let myToken = new ethers.Contract(myTokenAddress, LSP7Mintable.abi, signer);

Reviewer 2: Suggest const since variable is unchanged afterwards

```suggestion
  const myToken = new ethers.Contract(myTokenAddress, LSP7Mintable.abi, signer);
```

>   let mintTxn = await myToken.mint(signer.address, 1, true, '0x', { gasLimit: 400_000 })

Reviewer 2: Same here

```suggestion
  const mintTxn = await myToken.mint(signer.address, 1, true, '0x', { gasLimit: 400_000 })
```
